tests/integration_tests/security/test_spectre_meltdown.py

```diff
@@ -102,3 +210,87 @@ def run_spectre_meltdown_checker_on_guest(
     microvm.ssh.scp_file(spectre_meltdown_checker, remote_path)
     ecode, stdout, stderr = microvm.ssh.execute_command(f"sh {remote_path} --explain")
     assert ecode == 0, f"stdout:\n{stdout.read()}\nstderr:\n{stderr.read()}\n"
+
+
+def check_vulnerabilities_files_on_guest(microvm):
+    """
+    Check that the guest's vulnerabilities files do not contain `Vulnerable`.
+    See also: https://elixir.bootlin.com/linux/latest/source/Documentation/ABI/testing/sysfs-devices-system-cpu
+    and search for `vulnerabilities`.
+    """
+    vuln_dir = "/sys/devices/system/cpu/vulnerabilities"
+    ecode, stdout, stderr = microvm.ssh.execute_command(
+        f"grep -r Vulnerable {vuln_dir}"
+    )
+    assert ecode == 1, f"stdout:\n{stdout.read()}\nstderr:\n{stderr.read()}\n"
+
+
+def run_microvm(test_microvm_with_spectre_meltdown, network_config):
+    """
+    Run a microVM.
+    """
+    microvm = test_microvm_with_spectre_meltdown
+    microvm.spawn()
+    microvm.basic_config()
+    tap, host_ip, guest_ip = microvm.ssh_network_config(network_config, "1")
+    microvm.start()
+
+    return microvm, tap, host_ip, guest_ip
+
+
+def run_microvm_with_template(
+    test_microvm_with_spectre_meltdown, network_config, cpu_template
+):
+    """
+    Run a microVM with a template.
+    """
+    microvm = test_microvm_with_spectre_meltdown
+    microvm.spawn()
+    microvm.basic_config()
+    resp = microvm.machine_cfg.put(
+        vcpu_count=2,
+        mem_size_mib=256,
+        cpu_template=cpu_template,
+    )
+    assert microvm.api_session.is_status_no_content(resp.status_code)
+    tap, host_ip, guest_ip = microvm.ssh_network_config(network_config, "1")
+    microvm.start()
+
+    return microvm, tap, host_ip, guest_ip
```

Comment:
As these functions share some code, we might want to put them into one function like:
```suggestion
def run_microvm(
    test_microvm_with_spectre_meltdown, network_config, cpu_template=None
):
    """
    Run a microVM with a template.
    """
    microvm = test_microvm_with_spectre_meltdown
    microvm.spawn()
    microvm.basic_config()
    resp = microvm.machine_cfg.put(
        vcpu_count=2,
        mem_size_mib=256,
        cpu_template=cpu_template,
    )
    assert microvm.api_session.is_status_no_content(resp.status_code)
    tap, host_ip, guest_ip = microvm.ssh_network_config(network_config, "1")
    microvm.start()

    return microvm, tap, host_ip, guest_ip
```

Reply:
done